CMakeLists.txt

```diff
@@ -7,6 +7,22 @@ set( ${PROJECT_NAME}_VERSION_MINOR 5 )
 set( ${PROJECT_NAME}_VERSION_PATCH 0 )
 set( ${PROJECT_NAME}_VERSION  "${${PROJECT_NAME}_VERSION_MAJOR}.${${PROJECT_NAME}_VERSION_MINOR}.${${PROJECT_NAME}_VERSION_PATCH}" )
 
+# Define a default build type can be overriden by passing
+# ``-DCMAKE_BUILD_TYPE=<type>`` when invoking CMake
+if(NOT CMAKE_CONFIGURATION_TYPES)
+  if(NOT CMAKE_BUILD_TYPE)
+    set(CMAKE_BUILD_TYPE RelWithDebInfo
+      CACHE STRING "Choose the type of build, options are: None Release MinSizeRel Debug RelWithDebInfo"
+      FORCE
+      )
+  else()
+    set(CMAKE_BUILD_TYPE "${CMAKE_BUILD_TYPE}"
+      CACHE STRING "Choose the type of build, options are: None Release MinSizeRel Debug RelWithDebInfo"
+      FORCE
+      )
+  endif()
+endif()
+
```

Comment:
Is the else() part needed? The only thing it does is to cache it but CMAKE_BUILD_TYPE should be used by cmake regardless, no? 
Another comment is that we should do this key4hep-wide

Reply:
Good question. Maybe it should actually be for the first `if`? Tbh, I simply copied this from podio (I think).

Yes, this should definitely be done Key4hep wide. I think we already do in a few places, but definitely not all (and most importantly no in the project template).

Reply:
Ok. I'll open an issue somewhere to keep track of this and have a look at it for all the key4hep repos.